Assets/Countly/Plugins/CountlySDK/Services/ViewCountlyService.cs

```diff
@@ -25,198 +25,882 @@ private class ViewData
+                Log.Debug("[ViewCountlyService] AddSegmentationToViewWithName, consent is not given, ignoring the request.");
+                return;
+            }
+
+            lock (LockObj) {
+                Log.Info("[ViewCountlyService] Calling AddSegmentationToViewWithName for Name: [" + viewName + "]");
+
+                AddSegmentationToViewWithNameInternal(viewName, viewSegmentation);
+            }
+        }
+
+        /// <summary>
+        /// Updates the global segmentation
+        /// </summary>
+        /// <param name="viewSegmentation"></param>
+        public void UpdateGlobalViewSegmentation(Dictionary<string, object> viewSegmentation)
+        {
+            if (!_consentService.CheckConsentInternal(Consents.Views)) {
+                Log.Debug("[ViewCountlyService] UpdateGlobalViewSegmentation, consent is not given, ignoring the request.");
+                return;
+            }
+
+            lock (LockObj) {
+                Log.Info("[ViewCountlyService] Calling UpdateGlobalViewSegmentation sg[" + (viewSegmentation == null ? "null" : viewSegmentation.Count.ToString()) + "]");
+
+                if (viewSegmentation == null) {
+                    Log.Warning("[ViewCountlyService] When updating segmentation values, they can't be 'null'.");
                     return;
                 }
 
-                if (string.IsNullOrEmpty(name)) {
-                    return;
+                UpdateGlobalViewSegmentationInternal(viewSegmentation);
+            }
+        }
+        #endregion
+        #region InternalMethods
+        /// <summary>
+        /// Starts the view with given viewName, segmentation
+        /// </summary>
+        /// <param name="viewName">name of the view</param>
+        /// <param name="customViewSegmentation">segmentation that will be added to the view, set 'null' if none should be added</param>
+        /// <param name="viewShouldBeAutomaticallyStopped"></param>
+        /// <returns></returns>
+        private string StartViewInternal(string viewName, Dictionary<string, object> customViewSegmentation, bool viewShouldBeAutomaticallyStopped)
+        {
+            if (!_cly.IsSDKInitialized) {
+                Log.Warning("Countly.Instance.Init() must be called before startViewInternal");
+                return null;
+            }
+
+            if (_utils.IsNullEmptyOrWhitespace(viewName)) {
+                Log.Warning("[ViewCountlyService] StartViewInternal, Trying to record view with null or empty view name, ignoring request");
+                return null;
+            }
+
+            _utils.TruncateSegmentationValues(customViewSegmentation, _configuration.MaxSegmentationValues, "[ViewCountlyService] StartViewInternal, ", Log);
+            _utils.RemoveReservedKeysFromSegmentation(customViewSegmentation, reservedSegmentationKeysViews, "[ViewCountlyService] AutoCloseRequiredViews, ", Log);
+
+            int segmCount = 0;
```

Comment:
maybe for internal limits we could check whether or not dictionary is empty. If it is we can omit internal limits function calls

Reply:
yea that would reduce the call stack, good idea will apply